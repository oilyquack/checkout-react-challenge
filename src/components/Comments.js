import React from "react";
import Comment from "./Comment";

function Comments({ reviews }) {
  return (
    <div className="app__comments">
      {reviews.map((review, index) => {
        return <Comment key={index} review={review} />;
      })}
    </div>
  );
}

export default Comments;
