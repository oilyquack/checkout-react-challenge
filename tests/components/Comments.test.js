import React from "react";
import renderer from "react-test-renderer";
import Comments from "../../src/components/Comments";

it("matches the snapshot", () => {
  const reviews = [
    {
      name: "Connan Mockasin",
      comment: "Amazing!",
      rating: 5
    },
    {
      name: "Regina Spektor",
      comment: "Rubbish!",
      rating: 1
    }
  ];
  const tree = renderer.create(<Comments reviews={reviews} />);
  expect(tree).toMatchSnapshot();
});
