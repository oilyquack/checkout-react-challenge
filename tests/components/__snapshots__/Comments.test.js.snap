// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="app__comments"
>
  <div
    className="app__comments__comment"
  >
    <h2>
      Connan Mockasin
    </h2>
    ★★★★★
    <p>
      Amazing!
    </p>
  </div>
  <div
    className="app__comments__comment"
  >
    <h2>
      Regina Spektor
    </h2>
    ★☆☆☆☆
    <p>
      Rubbish!
    </p>
  </div>
</div>
`;
